### PR TITLE
ROC-3460: One app insights per product

### PR DESCRIPTION
**Pull request by rlewan**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-3460

### Change description ###

Adds App Insights and product-scoped Key Vault resources.

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!groovy
+//noinspection GroovyUnusedAssignment
+@Library("Infrastructure@feature/ROC-3460-One-app-insights-per-product") _
```
Comment (rlewan) [inline Jenkinsfile_CNP]: Will update after approval.

File: Jenkinsfile_parameterized [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!groovy
+//noinspection GroovyUnusedAssignment
+@Library("Infrastructure@feature/ROC-3460-One-app-insights-per-product") _
```
Comment (rlewan) [inline Jenkinsfile_parameterized]: Will update after approval.

File: application-insights.tf [hunk 3]
```diff
@@ -0,0 +1,10 @@
+resource "azurerm_application_insights" "appinsights" {
+  name                = "${var.product}-appinsights-${var.env}"
```
Comment (timja) [inline application-insights.tf]: as discussed remove appinsights from the name please

File: main.tf [hunk 4]
```diff
@@ -0,0 +1,10 @@
+locals {
+  default_resource_group_name = "${var.product}-${var.env}"
+  resource_group_name         = "${var.resource_group_name != "" ? var.resource_group_name : local.default_resource_group_name}"
```
Comment (timja) [inline main.tf]: don't think we need this part

File: main.tf [hunk 5]
```diff
@@ -0,0 +1,5 @@
+# Create a resource group
```
Comment (timja) [inline main.tf]: is this needed?

**Review by timja — APPROVED** [review]

**Pull request merged**